core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -136,138 +141,537 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
+    }, s"Expected a 'Stable' group status, rows and valid values for consumer id / client id / host columns in describe results for group $group.")
   }
 
   @Test
-  def testDescribeExistingGroupWithNoMembersWithNewConsumer() {
+  def testDescribeMembersOfExistingGroup() {
     TestUtils.createOffsetsTopic(zkUtils, servers)
+
     // run one consumer in the group consuming from a single-partition topic
-    consumerGroupExecutor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+    addConsumerGroupExecutor(new ConsumerGroupExecutor(brokerList, 1, group, topic))
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", group)
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val service = addConsumerGroupService(new KafkaConsumerGroupService(opts))
+
+    TestUtils.waitUntilTrue(() => {
+        val (state, assignments) = service.collectGroupMembers(false)
+        state == Some("Stable") &&
+        assignments.isDefined &&
+        assignments.get.count(_.group == group) == 1 &&
+        assignments.get.filter(_.group == group).head.consumerId.exists(_.trim != ConsumerGroupCommand.MISSING_COLUMN_VALUE) &&
+        assignments.get.filter(_.group == group).head.clientId.exists(_.trim != ConsumerGroupCommand.MISSING_COLUMN_VALUE) &&
+        assignments.get.filter(_.group == group).head.host.exists(_.trim != ConsumerGroupCommand.MISSING_COLUMN_VALUE)
+    }, s"Expected a 'Stable' group status, rows and valid member information for group $group.")
+
+    val (state, assignments) = service.collectGroupMembers(true)
+    assertTrue(s"Expected a topic partition assigned to the single group member for group $group",
+        assignments.isDefined &&
+        assignments.get.size == 1 &&
+        assignments.get.head.assignment.isDefined &&
+        assignments.get.head.assignment.get.size == 1)
+  }
+
+  @Test
+  def testDescribeStateOfExistingGroup() {
+    TestUtils.createOffsetsTopic(zkUtils, servers)
 
+    // run one consumer in the group consuming from a single-partition topic
+    addConsumerGroupExecutor(new ConsumerGroupExecutor(brokerList, 1, group, topic, "org.apache.kafka.clients.consumer.RoundRobinAssignor"))
     val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", group)
     val opts = new ConsumerGroupCommandOptions(cgcArgs)
-    consumerGroupService = new KafkaConsumerGroupService(opts)
+    val service = addConsumerGroupService(new KafkaConsumerGroupService(opts))
 
     TestUtils.waitUntilTrue(() => {
-      val (state, _) = consumerGroupService.describeGroup()
-      state == Some("Stable")
-    }, "Expected the group to initially become stable.")
+        val state = service.collectGroupState()
+        state.isDefined &&
+        state.get.state == Some("Stable") &&
+        state.get.numMembers == Some(1) &&
+        state.get.assignmentStrategy == Some("roundrobin")
```

Comment:
Curiously, this is the only assertion we have in this test case on the assignment strategy. Is it significant that it uses round-robin or could it just as well be range?

Reply:
Fair enough. Maybe a comment or perhaps this should even be a separate test case.

Reply:
Sure, I'll create a separate unit test for this.